### ISSUE TITLE
Document needing ts-node

### ISSUE THREAD

**Issue by zkldi** [tapjs/tapjs#740]
Perhaps I'm missing something obvious - as of v15 I cannot get module-based typescript to work.

Running with the TS flag set and typescript installed, I get
```
SyntaxError: Cannot use import statement outside a module
```

Enabling `"type": "module" in package.json results in
```
Can't run my Node.js Typescript project TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension “.ts” for (file)
```

And according to SO, the fix for that is to remove "type": "module". This happens regardless of what tsconfig's opinions on the output of TS is (I have it set to CommonJS, but it doesn't change anything).

**Comment by zkldi** [comment]
Update: This can be fixed by manually installing ts-node (similar to how you have to now manually install typescript), but this is not documented anywhere. Can submit a pull request to add this to the docs in a bit.

A warning should possibly be emitted if the TS flag is set but ts-node is not present, as at the moment it silently defaults to JS loading.

**Comment by StarpTech** [comment]
@zkldi thank you! Complete solution is:

1. Install `typescript`, `ts-node`, `@types/tap`
2. Run `tap --ts`

There is a typo in the [docs](https://node-tap.org/docs/cli/) it's `--ts` not `-ts`.

**Comment by zkldi** [comment]
PR in #744

**Comment by zkldi** [comment]
Closed as #744 got merged.
